### PR TITLE
FIX-#1497: Don't sort in concat() when sort=False

### LINKED ISSUE #1497: concat ignores sort argument
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 19.04
- **Modin version** (`modin.__version__`): 0.7.3+52.g880545b
- **Python version**: Python 3.7.3
- **Code we can use to reproduce**:

### Describe the problem

sort argument in modin.pandas.concat calls doesn't affect the result. It works like sort is always True.

### Source code / logs

import modin.pandas as pd

a = pd.DataFrame({'c' : [3],
                  'd' : [4]},
                 columns=list('dc'))
b = pd.DataFrame({'a' : [1],
                  'b' : [2]},
                 columns=list('ba'))

print(pd.concat([a, b], sort=True))
print(pd.concat([a, b], sort=False))

### PR DESCRIPTION
**Pull request by itamarst**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1497 
- [x] tests added and passing

### CODE REVIEW

**Comment by itamarst** [comment]
Investigating the failing tests now.

**Comment by itamarst** [comment]
It appears that the decision about when to sort is highly inconsistent within Pandas :cry:

**Comment by itamarst** [comment]
The issue appears to be https://github.com/pandas-dev/pandas/issues/35092 — Pandas has bug where `append()` sorts even when it shouldn't.

What is the policy for matching Pandas bugs?

**Comment by devin-petersohn** [comment]
@itamarst We try to match pandas behavior when possible. Is Modin behaving as pandas is supposed to after your fixes? In that case, I would say that we can probably just throw a warning linking to the pandas bug until it is resolved and keep the correct behavior.

**Comment by itamarst** [comment]
For `append()`:

1. Pandas has a bug where `sort=False` is ignored.
2. Previously Modin matched Pandas' buggy behavior.
3. Modin now behaves the way Pandas should behave when the bug is fixed.

So warning is probably the way to go, but will also have to figure out how to make the `append()` test pass... do a manual sort, I guess.

**Comment by devin-petersohn** [comment]
@itamarst That sounds good. In the test, we can check if the result is not sorted, but should be before we sort the pandas side so that we don't accidentally hide any bugs in the future.

**Comment by itamarst** [comment]
OK, done hopefully.

File: modin/engines/base/frame/data.py [hunk 1]
```diff
@@ -896,30 +896,37 @@ def _join_index_objects(self, axis, other_index, how, sort):
...
-            axis : 0 or 1
-                The axis index object to join (0 - rows, 1 - columns).
-            other_index : Index
-                The other_index to join on.
-            how : {'left', 'right', 'inner', 'outer'}
-                The type of join to join to make.
-            sort : boolean
-                Whether or not to sort the joined index
+
+        axis : 0 or 1 The axis index object to join (0 - rows, 1 - columns).
```
Comment (devin-petersohn) [inline modin/engines/base/frame/data.py:905]: I think this was accidentally changed, please revert these documentation formatting changes.

Comment (itamarst) [reply modin/engines/base/frame/data.py]: Done.

**Pull request merged**